executor/src/witgen/generator.rs

```diff
@@ -114,23 +132,46 @@ impl<'a, T: FieldElement> Generator<'a, T> {
         first_row
     }
 
+    /// Sets the "default" operation ID in the given row.
+    /// This assumes that there is a fixed column `main.p_line` and a witness column
+    /// `main._operation_id`, which will be the case if the PIL has been compiled from
+    /// Powdr ASM. By convention, the operation ID is last line of the ROM, i.e., the
+    /// largest value in `main.p_line`.
+    fn set_default_operation_id(&self, row: &mut Row<'a, T>) {
+        let rom_line_column = self.fixed_data.fixed_column_by_name("main.p_line").unwrap();
+        let default_operation_id = rom_line_column.values.iter().max().unwrap();
```

Comment:
@chriseth I don't think there is any way auto-detect the default operation ID, we just have to assume that the last line in the ROM points to an infinite loop, or do you see a way?

The way it is now, witgen will work if either all cells are uniquely constrained (e.g. using the `_sigma` logic) or it is the exact mechanism used when powdr ASM is compiled (i.e., we assume a `p_line` and `_operation_id` column in the machine and that the last line is the default operation ID).

Do you have variations of this in mind that should also be supported? I'm not sure if there is a fully general way that would cover all possible mechanisms.